plugin.audio.cbcradio/cbc.py

```diff
@@ -0,0 +1,235 @@
+            news_break()
+            while calc_minutes() in [00, 1, 2, 3, 4, 5]:
+                chill(1)
+            now = time.time() * 1000
+            program.time_end = now - 10000  # make sure program gets changed after news
+            c = -1  # make sure program gets updated if no playlog
+        if program.time_end < now:
+            try:
+                program = now_playing.get_current_program(program_schedule)
+                playlog = now_playing.get_playlog(program, location)
+                set_program_art(program)
+            except Exception:
+                pass
+        if c == 12 and not playlog:  # if there's no playlog, check every min
+            try:
+                playlog = now_playing.get_playlog(program, location)
+            except Exception:
+                pass
+            finally:
+                c = 0
+        if playlog:  # if there is a playlog, get which track should be playing
+            track = now_playing.get_current_track(playlog)
+            if track != last_track:  # if it's a new track, update 'now playing'
+                last_track = track
+                play_item = PLAYER.getPlayingItem()
+                tag = play_item.getMusicInfoTag()
+                tag.setAlbum(track.album)
+                tag.setArtist(track.artist)
+                tag.setTitle(track.title)
+                tag.setComment(f"{program.title} with {program.host}")
+                PLAYER.updateInfoTag(play_item)
+                # could not get this to work for v19
+                # if someone knows how to update currently playing
+                # this could be added to the matrix repo as well
+                '''play_item = xbmcgui.ListItem()
+                play_item.setPath(PLAYER.getPlayingFile())
+                play_item.setInfo('music', {'album': track.album, 'artist': track.artist, 'title': track.title})
+                PLAYER.updateInfoTag(play_item)'''
+                xbmc.log(level=xbmc.LOGDEBUG, msg="plugin.audio.cbcradio: track updated")
+        elif program.time_end < now or c == -1:  # if program over or plugin just started
+            play_item = PLAYER.getPlayingItem()
+            tag = play_item.getMusicInfoTag()
+            title = tag.getTitle()
+            # if there is no playlog, we're going to set the info displayed to
+            # the program name and host. this is mainly for CBC One as most of
+            # the shows do not have playlogs. CBC Music, usually does, but
+            # they can take awhile to be posted. this will display the show/host
+            # until the playlog is available.
+            if title == program.title:
+                pass
+            else:
+                tag.setAlbum(None)
+                tag.setArtist(program.host)
+                tag.setTitle(program.title)
+                tag.setComment(None)
+                PLAYER.updateInfoTag(play_item)
+        chill(5)
+        if MONITOR.abortRequested():
+            abort()
```

Comment:
ah crap.... was going to do an abort func and then changed it to chill()
i'll fix this tomorrow.

Reply:
i guess have a look anyway and let me know if there is anything besides this to fix before i resubmit.

Reply:
actually, it should technically never reach the non-existent func. the check for `MONITOR.abortRequested()` is in the `chill()` func.... testing on my dev box i never get an exception when exiting.

up to you if you want it removed or not.